types/test_utils.go

```diff
@@ -25,32 +25,71 @@ func GetRandomValidatorSetWithPrivKey() (*cmtypes.ValidatorSet, ed25519.PrivKey)
 	}, privKey
 }
 
-func GetRandomSignedHeader() (*SignedHeader, ed25519.PrivKey, error) {
-	valSet, privKey := GetRandomValidatorSetWithPrivKey()
-	signedHeader := &SignedHeader{
-		Header: Header{
-			Version: Version{
-				Block: InitStateVersion.Consensus.Block,
-				App:   InitStateVersion.Consensus.App,
+func GetRandomBlock(height uint64, nTxs int) *Block {
+	block := &Block{
+		SignedHeader: SignedHeader{
+			Header: Header{
+				BaseHeader: BaseHeader{
+					Height: height,
+				},
+				Version:         Version{Block: InitStateVersion.Consensus.Block},
+				AggregatorsHash: make([]byte, 32),
+			}},
+		Data: Data{
+			Txs: make(Txs, nTxs),
+			IntermediateStateRoots: IntermediateStateRoots{
+				RawRootsList: make([][]byte, nTxs),
 			},
+		},
+	}
 
-			BaseHeader: BaseHeader{
-				ChainID: "test",
-				Height:  uint64(rand.Int63()), //nolint:gosec,
-				Time:    uint64(time.Now().UnixNano()),
-			},
-			LastHeaderHash:      GetRandomBytes(32),
-			LastCommitHash:      GetRandomBytes(32),
-			DataHash:            GetRandomBytes(32),
-			ConsensusHash:       GetRandomBytes(32),
-			AppHash:             GetRandomBytes(32),
-			LastResultsHash:     GetRandomBytes(32),
-			ProposerAddress:     valSet.Proposer.Address,
-			AggregatorsHash:     valSet.Hash(),
-			NextAggregatorsHash: valSet.Hash(),
+	block.SignedHeader.AppHash = GetRandomBytes(32)
+
+	for i := 0; i < nTxs; i++ {
+		block.Data.Txs[i] = GetRandomTx()
+		block.Data.IntermediateStateRoots.RawRootsList[i] = GetRandomBytes(32)
+	}
+
+	// TODO(tzdybal): see https://github.com/rollkit/rollkit/issues/143
+	if nTxs == 0 {
+		block.Data.Txs = nil
+		block.Data.IntermediateStateRoots.RawRootsList = nil
+	}
+
+	return block
+}
+
+func GetRandomHeader() Header {
+	return Header{
+		BaseHeader: BaseHeader{
+			Height:  12,
+			Time:    uint64(time.Now().Local().Day()),
```

Comment:
Do we know why this is `.Local().Day()`? This is setting the time to 1-31. 
Should this be `uint64(time.UnixNano())`?